CTF Page

AutinMitra opened the pull request:
Lots of dirty/copypasta code from presentations, will extract and clean things up tomorrow morning. CTFCard is not stylized properly, and will later need support for viewing tj members

AutinMitra committed:
Init CTF Page - needs a lot of work

AutinMitra committed:
Small Ctfcard improvements

AutinMitra committed:
Extract debounce + sorry gink

AutinMitra committed:
edges.node -> nodes

DarinMao committed:
small performance improvements

DarinMao committed:
we are professional react developers

AutinMitra committed:
Extract CardGrid

AutinMitra committed:
Added scoreboard

AutinMitra commented:
Scoreboard is a teensy bit hacky - not sure how to properly integrate with theme-ui since it is customized via inline-styles
Scoreboard may or may not be too skinny

DarinMao commented:
can we refactor the searchable grid into a component? not sure how it would work, maybe you can pass in a card component function that gets a prop or something

AutinMitra commented:
Yeah extracting search logic into a single component would be a good idea, although it may be a bit hacky since the two pages have different card components. Perhaps adding an extra prop specifying whether the Grid will contain ctfCards or lectureCards.

We could also pass in a list containing the data + components like `[{data: {lecture: "lecturename"}, component: <Card />}]` (this feels hacky tho).

We could also just React.cloneElement and spread our data into the component props, I *think* that should work.

There's a good chance I'll wake up tomorrow morning, and realize that there's a much much better & cleaner way to do this

AutinMitra commented on src/components/scoreboard.js:
```diff
@@ -0,0 +1,143 @@
+import theme from '../gatsby-plugin-theme-ui/index' 
+import Modal from 'react-modal'
+
+const modalStyles = {
```
Is there any inline-styling for theme-ui? Couldn't seem to find any docs regarding it

DarinMao replied:
wdym?

AutinMitra replied:
react-modal requires us to use some react inline-styling to customize the overlay and modal dialog, so we don't have direct access to theme-ui features.

Is there a way to define classes in the ThemeUI theme?

AutinMitra commented on src/pages/ctf.js:
```diff
@@ -0,0 +1,111 @@
+                mb: [3, null, 0],
+              }}
+              as='a'
+              href="https://ctf.tjcsec.club/" // Should probably extract to access via graphql?
```
How should we extract this?

DarinMao replied:
honestly it's not gonna change so idk if it needs to be

AutinMitra commented on src/pages/presentations.js:
```diff
@@ -8,79 +8,44 @@ import Hero from '../components/hero'
+  const fuse = useRef(new Fuse(lectures, fuseOptions)).current
 
-  const search = useCallback(debounce((value) => {
+  const search = useRef(debounce((value) => {
```
We can also maybe use something like lodash for our debouncing, and convert it into a custom hook if someone hasn't already (not necessary since the current debounce works properly)

DarinMao replied:
yeah lodash is fine, though I copied this debounce from some blog post so it's 100% reliable

AutinMitra commented on src/pages/presentations.js:
```diff
@@ -161,7 +126,18 @@ const Presentations = ({ data }) => {
-          <CardGrid lectures={displayedLectures} />
+          <CardGrid>
+            {displayedLectures.map((lecture) => (
+              <LectureCard
```
Subject to change when we extract search logic

AutinMitra committed:
Extract & add search functionality to cardgrid

DarinMao committed:
small changes

- fixed typo in lectures (SQL Inection)
- expanded min cardgrid col size for small screens
- stretch card buttons on small screens
- added subtitle to hero
- moved fuseOptions to module scope

DarinMao committed:
Merge remote-tracking branch 'origin/master' into feat/ctf-page

DarinMao committed:
Merge remote-tracking branch 'origin/master' into feat/ctf-page

AutinMitra committed:
Converted old json to yml

DarinMao committed:
chore: clean up scoreboard and ctf

DarinMao committed:
å

DarinMao committed:
fix: change modal position to fixed

DarinMao committed:
add year to TJCTF 2018

AutinMitra commented:
Oo ok looks like you fixed styling, I don't think there's anything else to implement feature-wise

DarinMao committed:
fix: change ctf fuse thres to 0.3

AutinMitra committed:
Add transition to scoreboard

DarinMao committed:
fix: hidden navbar does not block stuff

DarinMao commented:
modal fades in but not out because it unmounts as soon as it is dismissed

worth fixing or nah

looks like you just need `closeTimeoutMS` prop on modal

AutinMitra committed:
Added closing transition props

AutinMitra commented:
Added the closing transition, not sure if adding ``theme.transitionTimes.fadeTime`` to the theme was a good idea though

AutinMitra committed:
Rever theme customization and fix some padding

DarinMao committed:
add subtitle to pres page
